packages/react/src/primitives/Pagination/usePagination.ts

```diff
@@ -36,8 +36,8 @@ export const usePagination = (
     }
   }, [currentPage]);
 
-  const onChange = React.useCallback((newPage: number, _: number) => {
-    setCurrentPage(newPage);
+  const onChange = React.useCallback((newPageIndex?: number) => {
+    setCurrentPage(newPageIndex ?? 0);
```

Comment:
Maybe we could just not call `setCurrentPage` if `newPageIndex` is `undefined`? Seems like passing `0` could have unintended consequences, but totally defer to your judgement here

Reply:
Oh yea good call